src/eve/utils.py

```diff
@@ -289,6 +289,20 @@ def _decorator(base_cls: Type) -> Type:
     return _decorator
 
 
+def abc(original_class: type) -> type:
```

Comment:
The name of the function is confusing because it is not related with actual Abstract Base Classes. I think it should be renamed to something less ambiguous (e.g. `noninstantiable`).